add documentation for preview version support (#724)
digital-asset/docs.daml.com: docs/2.9.0/docs/canton/usermanual/versioning.rst

.. _release-version-to-protocol-version-table:

> (*) The protocol version marked with `*` is the preview protocol version. To use a preview protocol version, follow the  :ref:`documentation <how-do-i-enable-preview-features>`.

Reviewer: I'd use plural in this sentence because you can have several protocol versions.

> (*) The protocol versions marked with `*` is the preview protocol version. To use a preview protocol version, follow the  :ref:`documentation <how-do-i-enable-preview-features>`.

Reviewer 2: Can we replace "preview protocol version" with "Initial access protocol version"?

Reviewer: @hrischuk-da I agree with you that we should increase the terminology consistency. For the time being, I would prefer if we keep preview because this is aligned with the current code in Canton and Daml repository. Changing that means another round of PRs and synchronization across the two repos and currently I am not convinced that it is the best use of the resources.

In any case, we need to rediscuss the terminology for future versions in any way because "enable dev protocol versions" is not an accurate statement: dev is both a specific protocol version and a synonym of unstable. Should we schedule that for 3.0 or even 2.10 ?

> (*) The protocol version marked with `*` are the preview protocol versions. To use a preview protocol version, follow the  :ref:`documentation <how-do-i-enable-preview-features>`.

Reviewer 3: ```suggestion
(*) The protocol version marked with `*` is the preview protocol version. To use a preview protocol version, follow the  :ref:`documentation <how-do-i-enable-preview-features>`.
```
